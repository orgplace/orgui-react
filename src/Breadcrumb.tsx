import classNames from 'classnames';
import React from 'react';

class OuiBreadcrumbItem extends React.Component<React.AnchorHTMLAttributes<HTMLAnchorElement>, {}> {
  render() {
    return (
      <a {...this.props} className={classNames(
        'ouiBreadcrumbItem',
        this.props.className,
      )}>{this.props.children}</a>
    );
  }
}

interface OuiBreadcrumbAttributes {
  children?: React.ReactElement<OuiBreadcrumbItem>[] | React.ReactElement<OuiBreadcrumbItem>
}

const OuiBreadcrumb: React.FC<React.HTMLAttributes<HTMLElement> & OuiBreadcrumbAttributes> = (props) => (
  <nav {...props} className={classNames(
    'ouiBreadcrumb',
    props.className,
  )}>{props.children}</nav>
);

export { OuiBreadcrumb, OuiBreadcrumbItem };
