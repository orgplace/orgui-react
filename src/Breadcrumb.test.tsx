import React from 'react';
import renderer from 'react-test-renderer';
import { OuiBreadcrumb, OuiBreadcrumbItem } from './Breadcrumb';

test('OuiBreadcrumb', function() {
	const component = renderer.create(
		<OuiBreadcrumb>
			<OuiBreadcrumbItem href="#">A</OuiBreadcrumbItem>
			<OuiBreadcrumbItem href="#">B</OuiBreadcrumbItem>
			<OuiBreadcrumbItem href="#">C</OuiBreadcrumbItem>
		</OuiBreadcrumb>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
