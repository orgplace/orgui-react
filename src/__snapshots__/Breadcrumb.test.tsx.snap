// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuiBreadcrumb 1`] = `
<nav
  className="ouiBreadcrumb"
>
  <a
    className="ouiBreadcrumbItem"
    href="#"
  >
    A
  </a>
  <a
    className="ouiBreadcrumbItem"
    href="#"
  >
    B
  </a>
  <a
    className="ouiBreadcrumbItem"
    href="#"
  >
    C
  </a>
</nav>
`;
